### PR TITLE
fix: sonar issues - Unused function parameters

### PR DESCRIPTION
**Pull request by jcraiglo1**
https://redhat.atlassian.net/browse/AAP-77394

One of multiple PRs addressing sonar quality gate issues. Removes unused function parameters.

Attached is a document describing the actions taken here.

[sonarcloud-remediation-report.md](https://github.com/user-attachments/files/29103602/sonarcloud-remediation-report.md)

Assisted-by: Claude

## Summary by CodeRabbit

* **Chores / Refactor**
  * Clarified unused or internal method parameters across shared-resource validation, pod startup waiting, and revision parsing by renaming them without changing runtime behavior or error handling.

### CODE REVIEW

File: src/aap_eda/services/project/scm.py [hunk 1]
```diff
@@ -359,10 +359,10 @@ def __call__(
                 )
                 if match:
                     return match.group(1)
-            match = re.search(
-                r'fatal: \[localhost\]: FAILED! => \{.+"msg": (.+)\}',
-                outputs.getvalue(),
-            )
+            output = outputs.getvalue()
```
Comment (jcraiglo1) [inline src/aap_eda/services/project/scm.py]: This had to be done to comply with sonar. The other way to handle this would've been to make a pr just to get this in so that this PR can go in. Fun new world we live in 😃

**Review by mkanoor — DISMISSED** [review]

File: src/aap_eda/services/project/scm.py (hunk 1 above)
Comment (ttuffin) [reply src/aap_eda/services/project/scm.py]: Claude flagged this change with the following concerns:

  1. The new pattern [^}]+ stops at the first } it finds. The original used .+ (greedy, with dotall default off but single-line) across the whole \{.+\} structure. Real Ansible FAILED output can look like:
  fatal: [localhost]: FAILED! => {"changed": false, "msg": "some error"}
  1. Here [^}]+ before \} would match "some error" correctly. But if msg value itself contains a } (e.g., JSON or dict-like content in the message string), the new pattern truncates prematurely.
  2. The guard condition '"msg":' in output (note: no space) will match "msg": anywhere in the output, not just within the fatal error block. The original regex was anchored to the fatal: [localhost]: FAILED! => \{...\} structure.
  The new split check is less precise — it could be triggered by "msg": appearing in unrelated output lines and then try to extract from the wrong location.
  3. The PR description claims this was a "sonar unused parameter" fix. But the regex in __call__ has no unused parameters — this regex change appears unrelated to the S1172 issue and should not be bundled with parameter renames.
  It changes functional behavior and deserves its own PR and review.
  4. No tests cover GitAnsibleRunnerExecutor.__call__ directly — the unit tests mock the executor entirely and never exercise the real __call__ method's error-parsing logic. The regression risk is undetected by the test suite.

Risk: Medium. The new regex is more fragile for edge cases where msg values contain } characters or where "msg": appears in other output sections.

Comment (jcraiglo1) [reply src/aap_eda/services/project/scm.py]: This seems to be a bit of a mess so I am going to make a new PR for this change. I'll link it here when it's up and remove this regex update

Comment (jcraiglo1) [reply src/aap_eda/services/project/scm.py]: https://github.com/ansible/eda-server/pull/1613

Comment (jcraiglo1) [reply src/aap_eda/services/project/scm.py]: Removed this update

**Review by mkanoor — APPROVED** [review]

**Review by B-Whitt — APPROVED** [review]
